staging/src/k8s.io/client-go/discovery/discovery_client.go

```diff
@@ -170,7 +187,7 @@ func (d *DiscoveryClient) ServerResourcesForGroupVersion(groupVersion string) (r
 	resources = &metav1.APIResourceList{
 		GroupVersion: groupVersion,
 	}
-	err = d.restClient.Get().AbsPath(url.String()).Do().Into(resources)
+	err = d.restClient.Get().Timeout(timeout).AbsPath(url.String()).Do().Into(resources)
```

Comment:
wait, isn't this the wrong spot.  You need to set it for the context timeout/deadline, right?

Reply:
@deads2k Thanks, I believe the pieces needed to set a context timeout/deadline are being added in https://github.com/kubernetes/kubernetes/pull/53076/files#diff-063e73bab84834a4187b1ad4865050adR217